src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonElementParseTests.cs

```diff
@@ -225,5 +226,134 @@ public static void TryParseValueInvalidDataFail(string json)
 
             Assert.Equal(0, reader.BytesConsumed);
         }
+
+        [Theory]
+        [InlineData("null")]
+        [InlineData("\r\n    null ")]
+        [InlineData("false")]
+        [InlineData("true ")]
+        [InlineData("   42.0 ")]
+        [InlineData(" \"str\" \r\n")]
+        [InlineData(" \"string with escaping: \\u0041\\u0042\\u0043\" \r\n")]
```

Comment:
Would it make sense to add a test case to illustrate what you had described as a possible common pitfall -- that SequenceEqual cannot be used to establish equality in cases where one string has escapes and one string doesn't?  I don't know how much tests in dotnet/runtime are used to illustrate concepts, to if that's uncommon, maybe it doesn't make sense to do.

Reply:
It wouldn't be applicable to when testing the newly added method per se, but to give you an example using the highlighted test values according to [section 7 of the JSON RFC](https://datatracker.ietf.org/doc/html/rfc8259#section-7) the JSON strings
```json
"ABC"
```
and 
```json
"\u0041\u0042\u0043"
```
should be considered equal. If you parse the latter value to a `JsonElement` and then call `GetString()` on it, the implementation will make sure that you get back an unescaped version, namely `"ABC"`. That isn't the case when you're reading the raw string so if you did
```C#
("ABC"u8).SequenceEquals("\u0041\u0042\u0043"u8);
```
that would be returning false which might break your application. Assuming that the RHS is untrusted input, an attacker could potentially exploit deserialization logic that checks for the presence of a specific property in the payload like this.